code/modules/power/chem_combustion.dm

```diff
@@ -0,0 +1,610 @@
+		"dbreath" = 30,
+		"kerosene" = 25,
+		"firedust" = 20,
+		"napalm_goo" = 15,
+		"phlogiston" = 15,
+		"diethylamine" = 12,
+		"acetone" = 12,
+		"oil" = 10,
+		"fuel" = 8,
+		"pyrosium" = 7,
+		"ethanol" = 6,
+		"hydrogen" = 6,
+		"plasma" = 6,
+		"phosphorus" = 5,
+		"magnesium" = 5
+	) // kilowattage (returns about half for a default setup)
+
+	// bit wierd but a bunch of type checks feels bad
+	var/valid_tanks = list(
+		/obj/item/tank/air,
+		/obj/item/tank/oxygen,
+		/obj/item/tank/anesthetic
+	)
+
+	// tanks
+	var/obj/item/reagent_containers/food/drinks/fueltank/fuel_tank
+	var/obj/item/tank/inlet_tank
+
+	// images
+	var/image/fuel_tank_image
+	var/image/inlet_tank_image
+
+	New()
+		..()
+		if(!src.net_id)
+			src.net_id = generate_net_id(src)
+
+		src.AddComponent(/datum/component/packet_connected/radio, \
+			null,\
+			src.frequency, \
+			src.net_id, \
+			"receive_signal", \
+			FALSE, \
+			null, \
+			FALSE \
+		)
+
+	receive_signal(var/datum/signal/signal, receive_method, receive_param, connection_id)
+		if (!src.packet_control) // we are not accepting any packets
+			return
+
+		if (!signal || !src.net_id || signal.encryption)
+			return
+
+		var/sender = signal.data["sender"] // what we send responses to
+
+		if (signal.data["address_1"] == "ping")
+			src.post_status(sender, "command", "ping_reply", "netid", src.net_id, "device", src.device)
+			return
+
+		if (signal.data["address_1"] != src.net_id)
+			return
+
+		switch (signal.data["command"])
+			if ("status")
+				src.send_status(sender)
+
+			if ("set_var")
+				var/field = signal.data["var_name"]
+				var/data = signal.data["data"]
+				if (!field || !data)
+					return
+
+				switch (field)
+					if ("active")
+						data = text2num_safe(data)
+						if (data)
+							if (src.start_engine(FALSE))
+								src.post_status(sender, "command", "error", "data", "ENGINE_START_FAILED", multicast = 1)
+								return
+
+							src.send_status(sender)
+							src.updateIntDialog()
+
+						else
+							src.stop_engine(FALSE)
+							src.send_status(sender)
+							src.updateIntDialog()
+
+					if ("air_inlet")
+						data = clamp(text2num_safe(data), 0, 100)
+						if (isnull(data))
+							return
+
+						if (data) data /= 100
+
+						atmos_inlet = data
+						src.send_status(sender)
+						src.updateIntDialog()
+
+					if ("fuel_inlet")
+						data = clamp(text2num_safe(data), 0, 100)
+						if (isnull(data))
+							return
+
+						if (data) data /= 100
+
+						fuel_inlet = data
+						src.send_status(sender)
+						src.updateIntDialog()
+
+					else
+						src.post_status(sender, "command", "error", "data", "INVALID_FIELD", multicast = 1)
+
+	update_icon()
+		if (src.active)
+			if (src.last_output >= 40000)
+				src.icon_state = "chemportgen3"
+			else if (src.last_output >= 20000)
+				src.icon_state = "chemportgen2"
+			else
+				src.icon_state = "chemportgen1"
+		else
+			src.icon_state = "chemportgen0"
+
+		src.ClearSpecificOverlays("fueltank", "inlettank")
```

Comment:
There a reason to clear and update each time? Typically you can just update and clear when applicable.